Server/src/client_thread.cpp

```diff
@@ -94,12 +97,17 @@ void *ClientThread(void *arg) {
     pthread_cleanup_push(ClientThread_cancel, &cInfo);
 
     try {
+#ifndef REDIS_ENABLED
         // connect to message bus
         cInfo.mbus = new msgBus_kafka(logger, thr->cfg, thr->cfg->c_hash_id);
 
         if (thr->cfg->debug_msgbus)
             cInfo.mbus->enableDebug();
-
+#else
+        // connect to redis
+        cInfo.redis = std::make_shared<MsgBusImpl_redis>(logger, thr->cfg, cInfo.client);
+        cInfo.redis->ResetAllTables();
```

Comment:
code in sonic-swss-common is not thread safe.
So, if there are multiple thread access to same object of sonic-swss-common class, you need add code to MsgBusImpl_redis to avoid race condition.

Reply:
This should be fine, we only support single connection from FRR to BMP, for multiple asic FRR/BMP should also be 
pair under the same namespace docker.

Reply:
#closed